pycbc/events/cuts.py

```diff
@@ -0,0 +1,336 @@
+                             "PARAMETER:VALUE:LIMIT, where, PARAMETER is the "
+                             "parameter to be cut, VALUE is the value at "
+                             "which it is cut, which must be convertible to "
+                             "a float, and LIMIT is one of '"
+                             + "', '".join(ineq_choices) +
+                             "' to indicate the inequality needed. "
+                             "PARAMETER is one of:'"
+                             + "', '".join(trigger_param_choices) +
+                             "'. For example snr:6:LOWER places the "
+                             "requirement that matched filter SNR must be > 6")
+    parser.add_argument('--template-cuts', nargs='+',
+                        help="Cuts to apply to the triggers, supplied as "
+                             "PARAMETER:VALUE:LIMIT. Format is the same as in "
+                             "--trigger-cuts. PARAMETER can be one of '"
+                             + "', '".join(template_param_choices) + "'.")
+
+
+# What are the inequalities associated with the cuts?
+# 'upper' means upper limit, and so requires value < the threshold
+ineq_functions = {
+    'upper': np.less,
+    'lower': np.greater,
+    'upper_inc': np.less_equal,
+    'lower_inc': np.greater_equal
+}
+
+
+def convert_inputstr(inputstr, choices):
+    """
+    Convert the inputstr into a dictionary keyed on parameter
+    with a tuple of the function to be used in the cut, and
+    the float to compare to.
+    Do some checks as to whether the inputstr fits input requirements
+    """
+    try:
+        cut_param, cut_value_str, cut_limit = inputstr.split(':')
+    except ValueError as value_e:
+        logging.warning("ERROR: Cut string format not correct, please "
+                        "supply as PARAMETER:VALUE:LIMIT")
+        raise value_e
+
+    if cut_param.lower() not in choices:
+        raise NotImplementedError("Cut parameter " + cut_param.lower() + " "
+                                  "not recognised, choose from "
+                                  + ", ".join(choices))
+    if cut_limit.lower() not in ineq_choices:
+        raise NotImplementedError("Cut inequality " + cut_limit.lower() + " "
+                                  "not recognised, choose from "
+                                  + ", ".join(ineq_choices))
+
+    try:
+        cut_value = float(cut_value_str)
+    except ValueError as value_e:
+        logging.warning("Error: Cut value must be convertible into a float, "
+                        "got '%s', a %s.", cut_value, type(cut_value))
+        raise value_e
+
+    return {cut_param: (ineq_functions[cut_limit],
+                        float(cut_value_str))}
+
+
+def ingest_cuts_option_group(args):
+    """
+    Check that the inputs given to options in insert_cuts_option_group
+    are sensible, and return the objects used to handle the cuts.
```

Comment:
I don't think this function does input checking, it just does some housekeeping between trigger and template cut options. So remove comment about check sensible inputs & say eg 'return dicts for trigger and template cuts' ?